src/app/courierPortal/page.js

```diff
@@ -0,0 +1,44 @@
+'use client';
+import styles from "/src/styles/style";
+import {Footer, Navbar} from "@/components";
+import {redirect} from "next/navigation";
+import React, {useEffect} from "react";
+import {useAuthState} from "react-firebase-hooks/auth";
+import {auth} from "@/firebase";
+
+const Page = () => {
+
+    const [user, loading] = useAuthState(auth);
+
+    useEffect(() => {
+        if (!loading && !user || !loading && window.localStorage.getItem('role') !== 'Courier' && window.localStorage.getItem('role') !== null) {
+            console.log('test' + window.localStorage.getItem('role'));
+            redirect('/')
+        }
+    }, [user, loading]);
```

Comment:
Here useEffect runs after the page is opened and rendered, even if it is a fraction of a section that the content will be visible before they get redirected, is there a better way to do this?

Reply:
anyways, i put in measure for userPortal where they can't obtain any information on the db other than their own orders so its fine, ill approve.